### DISCUSSION TITLE
How to use `--grammar` and `use-grammars`

### DISCUSSION THREAD

**Discussion by CptPotato** [helix-editor/helix#1817] [Q&A] [answered]
So I tried using the new `--grammar` option introduced in #1659. The PR mentions that it's possible to select which grammars to fetch/build by adding something like this to the top of `languages.toml`:

```toml
use-grammars = { only = [ "rust", "markdown" ] }
```

I tried doing that but it doesn't seem to work. I'm getting this error when calling `hx --grammar build` or `fetch`:
```
Error: data did not match any variant of untagged enum GrammarSelection for key 'use-grammars'
```

---

If I change it to this, the build and fetch commands seem to work:

```toml
use-grammars = [ "rust", "markdown" ]
```

So.. how do you actually use the `use-grammars` config correctly? There doesn't seem to be any info in the docs, yet.

**Comment by the-mikedavis** [comment, marked as answer]
It looks like there was a bug was preventing the `use-grammars` value from being deserialized as expected (see https://github.com/helix-editor/helix/pull/1818).

As for docs, the doc publishing setup was recently changed to show the latest release docs and master docs separately (see https://github.com/helix-editor/helix/pull/1783), so the new documentation for this lives [here](https://docs.helix-editor.com/master/languages.html#tree-sitter-grammars).

**Reply by CptPotato** [reply]
Thanks for the quick reply! this pretty much answers my question.

Also good to know that the docs are split now 👍
